apps/ex_nvr/lib/ex_nvr/devices/lpr_events_puller.ex

```diff
@@ -0,0 +1,55 @@
+defmodule ExNVR.Devices.LPREventsPuller do
+  use GenServer
+
+  require Logger
+
+  alias ExNVR.Devices.CameraClient
+  alias ExNVR.Events
+  alias ExNVR.Model.Device
+
+  @pulling_interval :timer.seconds(10)
+
+  def start_link(options) do
+    GenServer.start_link(__MODULE__, options)
+  end
+
+  @impl true
+  def init(options) do
+    Logger.metadata(device_id: options[:device].id)
+    Logger.info("Start LPR event puller")
+    device = options[:device]
```

Comment:
This should be the first statement and use it in all places instead of `options|:device]`. Or better do a pattern matching on the `init` function.